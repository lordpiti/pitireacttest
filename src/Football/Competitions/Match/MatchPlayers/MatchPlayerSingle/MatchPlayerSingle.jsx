import React from 'react';

const MatchPlayerSingle = ( props ) => {

  let substitutionIn, substitutionOut = null;

  if (props.player.substitutionIn) {
    substitutionIn = <span>
      <span className="glyphicon glyphicon-arrow-left"></span>
      <span>{props.player.substitutionIn.minute}</span>
    </span>;
  }

  if (props.player.substitutionOut) {
    substitutionOut = <span>
      <span className="glyphicon glyphicon-arrow-right"></span>
      <span>{props.player.substitutionOut.minute}</span>
    </span>;
  }

  return (
    <div className="row match-player-single">
    <div className="col-sm-1">{props.player.dorsal}</div>
    <div className="col-sm-5">
      {props.player.name} {props.player.surname}
    </div>
    <div className="col-sm-1">
        {substitutionIn}
        {substitutionOut}
    </div>
    <div className="col-sm-2">
      {props.player.bookings.map((booking, index) => {
        const cardImageUrl = booking.type==='Amarilla'?'/assets/img/yellow_card-512.png':'/assets/img/red_card-512.png';

        return <div key={index}>
          <img src={cardImageUrl} width="20" height="20" alt=""/>
          <label className="small">{booking.minute}</label>
        </div>
      }
      )}
    </div>
    <div className="col-sm-2">
      {props.player.goals.map((goal, index) =>
        <span key={index}>
          <img src="/assets/img/ball.png" width="20" height="20" alt=""/>
          <label className="small">{goal.minute}</label>
        </span>)
      }
    </div>
  </div>
  )
};

export default MatchPlayerSingle;