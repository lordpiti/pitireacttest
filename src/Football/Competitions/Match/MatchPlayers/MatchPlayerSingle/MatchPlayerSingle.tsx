import React from 'react';

interface PlayerMatchSingle {
  name: string;
  surname: string;
  dorsal: string;
  substitutionIn: Substitution;
  substitutionOut: Substitution;
  bookings: Booking[];
  goals: Goal[];
}

interface Booking {
  minute: number;
  type: string;
}

interface Goal {
  minute: number;
}

interface MatchPlayerSingleProps {
  player: PlayerMatchSingle;
}

interface Substitution {
  minute: number;
}

const MatchPlayerSingle = (props: MatchPlayerSingleProps) => {
  return (
    <div className='row match-player-single'>
      <div className='col-sm-1'>{props.player.dorsal}</div>
      <div className='col-sm-5'>
        {props.player.name} {props.player.surname}
      </div>
      <div className='col-sm-1'>
        {props.player.substitutionIn && (
          <span>
            <span className='glyphicon glyphicon-arrow-left'></span>
            <span>{props.player.substitutionIn.minute}</span>
          </span>
        )}
        {props.player.substitutionOut && (
          <span>
            <span className='glyphicon glyphicon-arrow-right'></span>
            <span>{props.player.substitutionOut.minute}</span>
          </span>
        )}
      </div>
      <div className='col-sm-2'>
        {props.player.bookings.map((booking, index) => {
          const cardImageUrl =
            booking.type === 'Amarilla'
              ? '/assets/img/yellow_card-512.png'
              : '/assets/img/red_card-512.png';

          return (
            <div key={index}>
              <img src={cardImageUrl} width='20' height='20' alt='' />
              <label className='small'>{booking.minute}</label>
            </div>
          );
        })}
      </div>
      <div className='col-sm-2'>
        {props.player.goals.map((goal, index) => (
          <span key={index}>
            <img src='/assets/img/ball.png' width='20' height='20' alt='' />
            <label className='small'>{goal.minute}</label>
          </span>
        ))}
      </div>
    </div>
  );
};

export default MatchPlayerSingle;
