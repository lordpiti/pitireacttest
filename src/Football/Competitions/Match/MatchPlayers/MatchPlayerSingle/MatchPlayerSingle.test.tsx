import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import MatchPlayerSingle, { PlayerMatchSingle } from './MatchPlayerSingle';

const playerData = {
  name: 'test name',
  surname: 'test surname',
  dorsal: '9',
  bookings: [
    {
      type: 'Amarilla',
      minute: 10,
    },
  ],
  goals: [
    {
      minute: 76,
    },
  ],
  substitutionIn: {
    minute: 9,
  },
  substitutionOut: {
    minute: 19,
  },
} as PlayerMatchSingle;

// https://www.freecodecamp.org/news/8-simple-steps-to-start-testing-react-apps-using-react-testing-library-and-jest/

afterEach(cleanup);
describe('MatchPlayerSingle', () => {
  it('renders correctly a player component with the right data', () => {
    const { asFragment } = render(<MatchPlayerSingle player={playerData} />);
    expect(asFragment()).toMatchSnapshot();
  });

  // it('should render the name and surname properly', () => {
  //   const { getByTestId } = render(<MatchPlayerSingle player={playerData} />);
  //   expect(getByTestId('playerNameSurname')).toBe(
  //     `${playerData.name} ${playerData.surname}`
  //   );
  // });
});
