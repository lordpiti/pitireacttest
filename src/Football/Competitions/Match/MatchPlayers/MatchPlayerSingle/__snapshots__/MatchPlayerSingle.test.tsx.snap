// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchPlayerSingle renders correctly a player component with the right data 1`] = `
<DocumentFragment>
  <div
    class="row match-player-single"
  >
    <div
      class="col-sm-1"
    >
      9
    </div>
    <div
      class="col-sm-5"
      data-testid="playerNameSurname"
    >
      test name test surname
    </div>
    <div
      class="col-sm-2"
    >
      <span>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-angle-double-left fa-w-14 "
          data-icon="angle-double-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
            fill="currentColor"
          />
        </svg>
        <label
          class="small"
        >
          9
        </label>
      </span>
      <span>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-angle-double-right fa-w-14 "
          data-icon="angle-double-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
            fill="currentColor"
          />
        </svg>
        <label
          class="small"
        >
          19
        </label>
      </span>
    </div>
    <div
      class="col-sm-2"
    >
      <div>
        <img
          alt=""
          height="20"
          src="/assets/img/yellow_card-512.png"
          width="20"
        />
        <label
          class="small"
        >
          10
        </label>
      </div>
    </div>
    <div
      class="col-sm-2"
    >
      <span>
        <img
          alt=""
          height="20"
          src="/assets/img/ball.png"
          width="20"
        />
        <label
          class="small"
        >
          76
        </label>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
